src/config-import.js

```diff
@@ -0,0 +1,231 @@
+          warnings.push(new Error(`Invalid field ${name}`))
+          continue
+        }
+        yield {
+          name,
+          value: fieldValue,
+        }
+      }
+    },
+
+    /**
+     * @returns {Iterable<{ fieldNames: string[], iconName: string | undefined, value: import('@mapeo/schema').PresetValue }>}
+     */
+    *presets() {
+      const { presets } = presetsFile
+      // sort presets using the sort field, turn them into an array
+      const sortedPresets = Object.keys(presets)
+        .map((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          if (!preset.sort) {
+            // if there's no sort field, put a big value - puts the field at the end -
+            preset.sort = 100
+          }
+          return preset
+        })
+        .sort((preset, nextPreset) => nextPreset.sort - preset.sort)
+
+      // 5. for each preset get the corresponding fieldId and iconId, add them to the db
+      for (let preset of sortedPresets) {
+        if (typeof preset !== 'object' || preset === null) {
+          warnings.push(new Error(`Invalid preset ${preset.name}`))
+          continue
+        }
+        /** @type {any} */
+        const presetValue = {
+          schemaName: 'preset',
+          fieldIds: [],
+          addTags: {},
+          removeTags: {},
+          terms: [],
+        }
+        for (const key of Object.keys(valueSchemas.preset.properties)) {
+          if (key in preset) {
+            // @ts-ignore - we validate below
+            presetValue[key] = preset[key]
+          }
+        }
+        if (!validate('preset', presetValue)) {
+          warnings.push(new Error(`Invalid preset ${preset.name}`))
+          continue
+        }
+        yield {
+          fieldNames:
+            'fields' in preset && Array.isArray(preset.fields)
+              ? preset.fields
+              : [],
+          iconName:
+            'icon' in preset && typeof preset.icon === 'string'
+              ? preset.icon
+              : undefined,
+          value: presetValue,
+        }
+      }
+    },
+  }
+}
+
+/**
+ * @param {string} filename
+ * @param {Buffer} buf
+ * @returns {{ name: string, variant: IconData['variants'][Number] }}}
+ */
+function parseIcon(filename, buf) {
+  const matches = filename.match(
+    /^([a-zA-Z0-9-]+)-([a-zA-Z0-9]+)@(\d+x)\.(png|jpg|jpeg)$/
+  )
+  if (!matches) {
+    throw new Error(`Unexpected icon filename ${filename}`)
+  }
+  /* eslint-disable no-unused-vars */
+  const [_, name, size, pixelDensity, extension] = matches
+  const density = Number(pixelDensity.replace('x', ''))
+  if (!(density === 1 || density === 2 || density === 3)) {
+    throw new Error(`Error loading icon. invalid pixel density ${density}`)
+  }
+  if (!(size === 'small' || size === 'medium' || size === 'large')) {
+    throw new Error(`Error loading icon. invalid size ${size}`)
+  }
+  if (!name) {
+    throw new Error('Error loading icon. missing name')
```

Comment:
I think we can remove this assertion because the regex matches one or more occurrences of any alfanum char, so name being empty would be caught by the first assertion (`if(!matches)`). But maybe there's a case I'm missing?

Reply:
I agree that it's redundant for now, but I'd prefer to keep this in case we want to enable [unchecked indexed access][0] in the future.

[0]: https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess